source/modules/MEI.shared.xml

```diff
@@ -4922,58 +4877,32 @@
       </attDef>
     </attList>
   </elementSpec>
-  <elementSpec ident="composer" module="MEI.shared">
-    <desc>The name of the creator of the intellectual content of a musical work.</desc>
-    <classes>
-      <memberOf key="att.common"/>
-      <memberOf key="att.bibl"/>
-      <memberOf key="att.evidence"/>
-      <memberOf key="att.facsimile"/>
-      <memberOf key="att.lang"/>
-      <memberOf key="model.respLikePart"/>
-    </classes>
-    <content>
-      <rng:zeroOrMore>
-        <rng:choice>
-          <rng:text/>
-          <rng:ref name="model.textPhraseLike.limited"/>
-        </rng:choice>
-      </rng:zeroOrMore>
-    </content>
-  </elementSpec>
-  <elementSpec ident="contributor" module="MEI.shared">
-    <desc>Names of individuals, institutions, or organizations responsible for contributions to the
-      intellectual content of a work, where the specialized elements for authors, editors, etc. do
-      not suffice or do not apply.</desc>
+  <elementSpec ident="agent" module="MEI.shared">
+    <desc>The name of an individual or corporate body who or which is related to any other entity. 
+      The nature of the relationship should be expressed within the <att>role</att> attribute.</desc><!-- TODO: check if speaker,audience, distributor are needed-->
     <classes>
+      <memberOf key="att.authorized"/>
+      <memberOf key="att.canonical"/>
       <memberOf key="att.common"/>
       <memberOf key="att.bibl"/>
       <memberOf key="att.evidence"/>
       <memberOf key="att.facsimile"/>
       <memberOf key="att.lang"/>
+      <memberOf key="model.itemLike"/>
+      <memberOf key="model.manifestationLike"/>
       <memberOf key="model.respLikePart"/>
     </classes>
     <content>
       <rng:zeroOrMore>
         <rng:choice>
           <rng:text/>
-          <rng:ref name="model.textPhraseLike.limited"/>
+          <rng:ref name="model.nameLike.agent"/> <!-- TODO: check if all elements are necessary-->
         </rng:choice>
       </rng:zeroOrMore>
     </content>
-    <constraintSpec ident="check_contributor_role" scheme="isoschematron">
-      <constraint>
-        <sch:rule context="mei:contributor">
-          <sch:assert
-            test="not(matches(normalize-space(lower-case(@role)),               '(arranger|author|composer|contributor|editor|funder|librettist|lyricist|sponsor)'))"
-            >The value of @role must not contain the name of another element available in this
-            context.</sch:assert>
-        </sch:rule>
-      </constraint>
-    </constraintSpec>
     <attList>
-      <attDef ident="role" usage="rec">
-        <desc>Used to specify the contributor's function.</desc>
+      <attDef ident="role" usage="req">
```

Comment:
Very few things in MEI are actually required. I would suggest this be `rec` (it is possible to have an agent and to not know what role they played!)

Reply:
My understanding was that `agent` would be an element on the highest level of abstraction. So it seemed sensible not only to point out that there is an individual or group of individuals somehow related to e.g. a work but also to specify the nature of the relation. If one doesn't know the nature of the relation it should still be possible to insert free text for the `role` attribute denoting that the relation is unknown otherwise it should be considered best practice to use a relation specified e.g. in the marc relators list. Does this make sense?
However, I think it might be a good starting point for discussion within the community, since there might are some use cases where requiring the `role` attribute might not be ideal.